sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/StorageApiWriteUnshardedRecords.java

```diff
@@ -745,15 +745,22 @@ long flush(
                 quotaError = statusCode.equals(Status.Code.RESOURCE_EXHAUSTED);
               }
 
+              int allowedRetry;
+
               if (!quotaError) {
                 // This forces us to close and reopen all gRPC connections to Storage API on error,
                 // which empirically fixes random stuckness issues.
                 invalidateWriteStream();
+                allowedRetry = 5;
+              } else {
+                allowedRetry = 10;
```

Comment:
originally, it will retry 5 times (make API call 6 times) within (1.5^6-1)/(1.5-1) ~ 20s. Now, 10 retries happened in (1.5^8-1)/(1.5-1) + 20/2 ~ 90s